packages/now-go/go-helpers.ts

```diff
@@ -127,50 +126,35 @@ export async function downloadGo(
   platform = process.platform,
   arch = process.arch
 ) {
-  // Check default `Go` in user machine
-  const isUserGo = await pathExists(join(homedir(), 'go'));
+  // Check if `go` is already installed in user's `$PATH`
+  const { failed, stdout } = await execa('go', ['version'], { reject: false });
 
-  // If we found GOPATH in ENV, or default `Go` path exists
-  // asssume that user have `Go` installed
-  if (isUserGo || process.env.GOPATH !== undefined) {
-    const { stdout } = await execa('go', ['version']);
+  if (!failed && parseInt(stdout.split('.')[1]) >= 11) {
+    debug('Using system installed version of `go`: %o', stdout.trim());
+    return createGo(dir, platform, arch);
+  }
 
-    if (parseInt(stdout.split('.')[1]) >= 11) {
-      return createGo(dir, platform, arch);
-    }
+  // Check `go` bin in builder CWD
+  const isGoExist = await pathExists(join(dir, 'bin'));
+  if (!isGoExist) {
+    debug('Installing `go` v%s to %o for %s %s', version, dir, platform, arch);
+    const url = getGoUrl(version, platform, arch);
+    debug('Downloading `go` URL: %o', url);
+    const res = await fetch(url);
 
-    throw new Error(
-      `Your current ${stdout} doesn't support Go Modules. Please update.`
-    );
```

Comment:
I don't see this error anymore:

```
Your current ${stdout} doesn't support Go Modules. Please update.
```

Don't we still need to throw if an old version of Go was found?

Reply:
If the system `go` binary is too old then this function will now fall back to downloading the requested `go` binary, basically the same behavior as if no system `go` binary was installed.